users/views.py

```diff
@@ -0,0 +1,188 @@
+"""
+This module contains
+"""
+# pylint: disable= no-self-use, no-member
+from django.contrib.auth.hashers import check_password, make_password
+from django.contrib.auth.password_validation import validate_password
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+from django.db.utils import IntegrityError
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from rest_framework import serializers, status
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.authtoken.models import Token
+from rest_framework.decorators import authentication_classes, permission_classes
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .models import User
+
+
+class UserSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the User model
+    """
+
+    class Meta:
+        """
+        Tells the models about which fields of the model to include in parsed response/request json.
+        """
+
+        model = User
+        fields = ["email", "name", "is_ative", "is_staff", "date_joined"]
+
+
+class LoginUser(APIView):
+    """
+    Logs a User and returns a Token
+    """
+
+    def post(self, request):
+        """
+        This functions validates the credentials and logs a user in
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): A json object containing message, code and token
+        """
+        response = {
+            "message": "Wrong email or password",
+            "code": status.HTTP_400_BAD_REQUEST,
+        }
+        try:
+            email = request.POST["email"]
+            password = request.POST["password"]
+            user = get_object_or_404(User, email=email)
+            if check_password(password, user.password):
+                response["message"] = "Logged in successfully"
+                response["code"] = status.HTTP_200_OK
+                if not Token.objects.filter(user=user).exists():
+                    token = Token.objects.create(user=user)
+                else:
+                    token = Token.objects.get(user=user)
+                response["token"] = token.key
+        except KeyError:
+            response["message"] = "Bad Request"
+        except Http404:
+            response["message"] = "User does not exist"
```

Comment:
We are already raising this in `user = get_object_or_404(User, email=email)` so we don't need to raise it again.
Instead of this, we can return a bad request saying `There was an error processing your request` with a 400.